# HW 1
## Problem 4

In [11]:
# Some python setup first
import numpy as np
from numpy import sin, cos, tan, arctan2, sqrt

def quadratic(a, b, c):
    return ((-b + sqrt(b**2 - 4*a*c)) / (2*a),
            (-b - sqrt(b**2 - 4*a*c)) / (2*a))

Let's start with a sketch of the car traveling and braking.

![hw1_4](hw1_4.png)

And use variables for the various quantities:

$v_0 = 20$ m/s, $a = -10$ m/s, $t_R = 0.5$ s, $d=35$ m.

In [2]:
v0 = 20.
a = -10.
tR = 0.5
d = 35.

The problem can be solved graphically or with formulas.

### Graphical solution

We can draw a velocity vs time graph from the given information: velocity starts at $v_0$, remains constant during the reaction time and then decreases at a rate of $a$ until the car comes to a stop ($v = 0$).

![hw1_4a.png](hw1_4a.png)

After 0.5 s, the car is still going 20 m/s. One second later, the velocity decreased by -10 m/s$^2$ $\cdot$ 1 s, so it's 10 m/s, and another second later it decreased by another 10 m/s to zero, so the braking time $t_B$ can be easily seen to be 2 seconds. More generally, we could use the formulate for velocity at constant acceleration.

$$v(t) = v_0 + a t$$

After the braking time $t_B$, the car comes to a stop, that is
$$0 = v(t_B) = v_0 + at_B \Longrightarrow t_B = -\frac{v_0}{a}$$
and as you see in the following box t_B in fact comes out to be 2.0 s.

In [5]:
tB = - v0/a
tB

2.0

The total distance $x_f$ traveled during reaction time and braking is just the area under the graph above, that is, the sum of a rectangle and a triangle:
$$x_f = v_0 t_R + \frac{1}{2} v_0 t_B,$$
so we can plug in the numbers to find $x_f = 30$ m:

In [6]:
xf = v0 * tR + .5 * v0 * tB
xf

30.0

Since the deer was 35 m away, the car managed to stop in time with 5 m to spare.

# Analytic solution

Instead, I could solve the problem just using the formulas for motion at constant acceleration.

During the reaction time, velocity is constant (acceleration is zero), so the car travels a distance $x_R$
$$
x_R = v_0 t_R
$$

During the braking part, velocity drops until it hits zero, so we use $v(t) = v_0 + at$ just as above and find
$$
t_B = -\frac{v_0}{a}
$$
The distance traveled while braking $x_B$ can be found from
$$
x_B = x(t_B) = \frac{1}{2}at_B^2 + v_0t_B
$$
and then those distances can be added up to find the total distance traveled 
$$
x_f = x_R + x_B
$$

Plugging in $x_R, x_B$, we find
$$
x_f = v_0 t_R + \frac{1}{2} a t_B^2 + v_0 t_B
$$

and finally plugging in $t_B$ as well
$$
x_f = v_0 t_R + \frac{1}{2} \frac{v_0^2}{a} - \frac{v_0^2}{a} = v_0 t_R - \frac{v_0^2}{2a}
$$

Plugging in numbers, we should get the same answer as before:

In [9]:
xf = v0 * tR - v0**2 / (2*a)
xf

30.0

As a by the way comment, in class you've derived the "3rd law of kinematics"
$$
v_f^2 - v_i^2 = 2 a \Delta x
$$
which can be used the find the braking distance directly, using $v_i = v_0$, $v_f = 0$, we get $x_B = \Delta x = - \frac{v_0^2}{2a}$ as well.

### Maximum speed to (just) not hit the deer

Since we already did our calculation with variables / formulas, find the max speed the car can go while still braking in time is not too bad. What we're looking for is: What is the $v_0$ such that $x_f = d = 35$ m?

$$
d = x_f = v_0 t_R - \frac{v_0^2}{2a}
$$
or
$$
- \frac{1}{2a}v_0^2 + t_R v_0 - d = 0
$$
where everything is known but $v_0$, so we can solve the quadratic equation $Av_0^2 + Bv_0 + C = 0$ with
$$
A = -\frac{1}{2a}, B = t_R, C = -d
$$

In [12]:
A = -1./(2*a)
B = tR
C = -d

In [13]:
v0_p, v0_m = quadratic(A, B, C)
v0_p, v0_m

(21.925824035672516, -31.925824035672516)

We need a positive initial velocity, so we found the maximum speed to be $21.9$ m/s. That answer is slightly larger than the 20 m/s which left 5 m to spare, which makes sense. (Maybe it's worth pointing out that just a very small increase in velocity of less than 2 m/s increased the braking distance enough to change the situation from fairly safe to borderline.)